pkg/storageos/deploy.go

```diff
@@ -351,13 +351,40 @@ func getPodNames(pods []corev1.Pod) []string {
 
 // GetNodeIPs returns a slice of IPs, given a slice of nodes.
 func GetNodeIPs(nodes []corev1.Node) []string {
-	var ips []string
+	ips := []string{}
 	for _, node := range nodes {
-		ips = append(ips, node.Status.Addresses[0].Address)
+		// Prefer InternalIP
+		if internalIP := GetNodeInternalIP(node.Status.Addresses); internalIP != "" {
```

Comment:
Is there any chance of this change breaking other (non-digitalocean) setups?

Reply:
I don't think so since InternalIP should always be set and used.  If for some reason InternalIP isn't set then we fall back to the previous behaviour (which probably wouldn't have worked anyways).

I've seen some k8s issues where multiple InternalIPs are set but this keeps the same behaviour (uses first).